e2e/src/pay-with-resume.spec.ts

```diff
@@ -0,0 +1,75 @@
+import { expect, test } from "@playwright/test";
+import {
+  getGraphQLClient,
+  setUpTestContext,
+  tearDownTestContext
+} from "./context";
+import payWithSendFlow from "./flows/pay-with-send-flow.json";
+import { fillGovUkCardDetails, navigateToPayComponent, waitForPaymentResponse } from "./pay.spec";
+import { fillInEmail } from "./save-and-return.spec";
+
+test.describe("Redirecting from Gov Pay in a Save & Return flow", () => {
+  const adminGQLClient = getGraphQLClient();
+  let context: any = {
+    user: {
+      firstName: "test",
+      lastName: "test",
+      email: "e2epaysendtest@test.com",
+    },
+    team: {
+      name: "Buckinghamshire",
+      slug: "buckinghamshire",
+      logo: "https://placedog.net/250/250",
+      primaryColor: "#F30415",
+      homepage: "example.com",
+    },
+    flow: {
+      slug: "e2e-pay-with-send-test-flow",
+      data: payWithSendFlow,
+    },
+    sessionIds: [],
+  };
+  const previewURL = `/${context.team.slug}/${context.flow.slug}/preview?analytics=false`;
+  const cards = {
+    successful_card_number: "4444333322221111",
+    invalid_card_number: "4000000000000002",
+  };
+
+  test.beforeAll(async () => {
+    try {
+      context = await setUpTestContext(context);
+    } catch (e) {
+      await tearDownTestContext(context);
+      throw e;
+    }
+  });
+
+  test.afterAll(async () => {
+    await tearDownTestContext(context);
+  });
+
+  test("redirecting back to the flow after a successful payment and skipping email confirmation", async ({
+    page,
+  }) => {
+    await page.goto(previewURL);
+    await fillInEmail({ page, context });
+    const sessionId = await navigateToPayComponent(page);
+    context.sessionIds.push(sessionId);
+
+    await page.getByText("Pay using GOV.UK Pay").click();
+    await fillGovUkCardDetails({
+      page,
+      cardNumber: cards.successful_card_number,
+    });
+    await page.locator("#confirm").click();
+    const { paymentId: actualPaymentId } = await waitForPaymentResponse(page);
+    await expect(page.getByText("Application sent")).toBeVisible();
+    await expect(page.getByText(actualPaymentId)).toBeVisible();
```

Comment:
I thought an e2e test was going to be the right way to test this intended redirect/resume behavior and proactively catch in the future, but then I naively realized after spinning up the test that the referrer is stripped from the browser when navigating from HTTPS to HTTP (ie localhost), therefore this e2e test fails for the same reason this behavior can't be tested in local development.

Any other ideas for how to test??

Reply:
Yeah this is a tricky one!

The only thing I can really see in the docs is setting this via the `Page.goto()` API (docs: https://playwright.dev/docs/api/class-frame#frame-goto-option-referer), but this doesn't really help us as we would need GovPay to set this.

I'd be ok to merge without the test and concentrate ideas on ways of avoiding `document.referrer` if possible.

Reply:
Thanks for thinking on this one too! Agree that sounds like ideal way forward and lets us fix bug soonest.

I've just updated this branch to remove the e2e commit and I'll just merge the single line change to add the new gov pay subdomain once CI finishes :+1: